server/src/test/java/org/elasticsearch/indices/recovery/DanglingIndicesIT.java

```diff
@@ -126,29 +146,291 @@ public Settings onNodeStopped(String nodeName) throws Exception {
      * Check that when dangling indices are not written, then they cannot be recovered into the cluster.
      */
     public void testDanglingIndicesAreNotRecoveredWhenNotWritten() throws Exception {
-        internalCluster().startNodes(3, buildSettings(false, true));
+        internalCluster().startNodes(3, buildSettings(0, false, true));
 
-        createIndex(INDEX_NAME, Settings.builder().put("number_of_replicas", 2).build());
-        ensureGreen(INDEX_NAME);
-        internalCluster().getInstances(IndicesService.class).forEach(
-            indicesService -> assertTrue(indicesService.allPendingDanglingIndicesWritten()));
+        createDanglingIndices(INDEX_NAME);
 
-        // Restart node, deleting the index in its absence, so that there is a dangling index to recover
+        // Since index recovery is async, we can't prove index recovery will never occur, just that it doesn't occur within some reasonable
+        // amount of time
+        assertFalse(
+            "Did not expect dangling index " + INDEX_NAME + " to be recovered",
+            waitUntil(() -> indexExists(INDEX_NAME), 1, TimeUnit.SECONDS)
+        );
+    }
+
+    /**
+     * Check that when dangling indices are discovered, then they can be listed.
+     */
+    public void testDanglingIndicesCanBeListed() throws Exception {
```

Comment:
Should we also test that this works if auto-import is enabled? You can make an un-importable dangling index by deleting it and creating another with the same name while the node is away.